src/main/java/uk/nhs/ctp/utils/ResourceUtils.java

```diff
@@ -0,0 +1,26 @@
+package uk.nhs.ctp.utils;
+
+import java.util.function.Predicate;
+import lombok.AccessLevel;
+import lombok.NoArgsConstructor;
+import org.hl7.fhir.dstu3.model.Reference;
+import org.hl7.fhir.dstu3.model.ResourceType;
+
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
+public class ResourceUtils {
+
+  public static Predicate<Reference> referenceTo(ResourceType resourceType) {
+    return reference -> {
+      if (reference.hasReferenceElement()) {
+        return reference.getReferenceElement().getResourceType().equals(resourceType.toString());
+      }
+      else if (reference.getResource() != null) {
```

Comment:
Minor point here, you can also remove the `else` keywords as the returns in the previous conditions imply this will only be reached as an `else`.